onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.cc

```diff
@@ -315,14 +311,12 @@ float GetQuantizationScale(const InitializedTensorSet& initializers, const Node&
 
 common::Status GetQuantizationZeroPoint(const InitializedTensorSet& initializers,
                                         const Node& node, size_t idx, int32_t& zero_point) {
-  std::unique_ptr<uint8_t[]> unpacked_tensor;
-  size_t tensor_byte_size;
+  std::vector<uint8_t> unpacked_tensor;
   const auto& zero_point_tensor = *initializers.at(node.InputDefs()[idx]->Name());
   ORT_RETURN_IF_ERROR(
-      onnxruntime::utils::UnpackInitializerData(zero_point_tensor, node.ModelPath(),
-                                                unpacked_tensor, tensor_byte_size));
+      onnxruntime::utils::UnpackInitializerData(zero_point_tensor, node.ModelPath(), unpacked_tensor));
   // Onnx quantization uses uint8 [int8 not yet supported], need to cast to int32_t used by NNAPI
-  zero_point = static_cast<int32_t>(unpacked_tensor.get()[0]);
+  zero_point = static_cast<int32_t>(unpacked_tensor[0]);
```

Comment:
>unpacked_tensor[0] [](http://example.com/codeflow?start=36&length=18)

Need to check that the data is not empty. Perhaps, this was the reason for crashes?

Reply:
Yes, we should check the length of the buffer, but this is not the reason for the crashes